lib/opentelemetry_tesla.ex

```diff
@@ -61,65 +61,71 @@ defmodule OpentelemetryTesla do
     )
   end
 
-  defp handle_start(_event, _measurements, _metadata, _config) do
+  defp handle_start(_event, _measurements, %{env: %Tesla.Env{method: method}} = metadata, _config) do
+    http_method = http_method(method)
+
     OpentelemetryTelemetry.start_telemetry_span(
       @tracer_id,
-      "External HTTP Request",
-      %{},
-      %{kind: :server}
+      "HTTP #{http_method}",
+      metadata,
+      %{kind: :client}
     )
   end
 
-  defp handle_stop(_event, %{duration: measurement}, metadata, _config) do
-    span_args =
-      metadata
-      |> headers_span_args()
-      |> :lists.append(span_args(metadata))
-      |> :lists.append([{"http.request.measurement", measurement}])
-
-    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, %{})
+  defp handle_stop(_event, _measurements, metadata, _config) do
+    span_attrs = build_attrs(metadata)
+    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, metadata)
 
-    OpenTelemetry.Span.set_attributes(ctx, span_args)
+    OpenTelemetry.Span.set_attributes(ctx, span_attrs)
 
-    OpentelemetryTelemetry.end_telemetry_span(@tracer_id, %{})
+    OpentelemetryTelemetry.end_telemetry_span(@tracer_id, metadata)
   end
 
   defp handle_exception(
          _event,
          %{duration: native_time},
-         %{kind: kind, reason: reason, stacktrace: stacktrace},
+         %{kind: kind, reason: reason, stacktrace: stacktrace} = metadata,
          _config
        ) do
-    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, %{})
+    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, metadata)
 
     exception = Exception.normalize(kind, reason, stacktrace)
 
     Span.record_exception(ctx, exception, stacktrace, duration: native_time)
     Span.set_status(ctx, OpenTelemetry.status(:error, ""))
-    OpentelemetryTelemetry.end_telemetry_span(@tracer_id, %{})
+    OpentelemetryTelemetry.end_telemetry_span(@tracer_id, metadata)
   end
 
-  defp headers_span_args(metadata) do
-    case Map.get(metadata, :env) do
-      nil ->
-        []
-
-      map ->
-        map
-        |> Map.get(:headers)
-        |> Enum.map(fn {key, value} -> {"http.headers.#{key}", value} end)
-    end
+  defp build_attrs(%{
+         env: %Tesla.Env{method: method, url: url, status: status_code, headers: headers}
+       }) do
+    uri = URI.parse(url)
+
+    attrs = [
+      "http.method": http_method(method),
+      "http.url": url,
+      "http.target": uri.path,
```

Comment:
How is Tesla handling query strings? Those are part of the target

Reply:
query strings are not directly "dumped" on `URL property of the event metadata, instead, they are in a separate property like this: `query: [token: "some-token"]` 

The option that I have at this moment is to use `URI.encode_query/1` to transform this into `token=some-token` and then append it to the original URL to end up with something like `http://foo.bar/path?token=some-token` and set `http.url` with this value. 

What do you think?